### ISSUE TITLE
No ads, no logs (Unity 2019.2.0, AdMob 3.18.2, Resolver 1.2.124)

### ISSUE THREAD

**Issue by iseeall** [googleads/googleads-mobile-unity#989]
After fixing this issue https://github.com/googleads/googleads-mobile-unity/issues/985 by importing 1.2.124 resolver from https://github.com/googlesamples/unity-jar-resolver/raw/v1.2.124/play-services-resolver-1.2.124.0.unitypackage (Resolver->Update does nothing btw and just keeps 1.2.123), the game builds but doesn't show any ads in runtime.

LogCat also has no messages. The only 2 I see are:
```
08-02 21:19:02.951: I/Ads(22858): Updating ad debug logging enablement.
08-02 21:19:02.971: W/Ads(22858): Update ad debug logging enablement as false
```
which probably disable logging? So it isn't even clear if there was any connection to the Google's servers or errors came from there.

Version Handler settings:

![VersionHandler](https://user-images.githubusercontent.com/8617651/62393843-a40db280-b56b-11e9-8522-94d4c173abe2.png)

Android resolver settings:

![AndroidResolverSettings](https://user-images.githubusercontent.com/8617651/62393862-b687ec00-b56b-11e9-9286-b7ebc7119b87.png)

Android Build settings:

![AndroidBuildSettings](https://user-images.githubusercontent.com/8617651/62393890-d7504180-b56b-11e9-9720-800fbda21693.png)

Publishing settings:

![PublishingSettings](https://user-images.githubusercontent.com/8617651/62393946-0a92d080-b56c-11e9-8eb7-85415c64e063.png)

Other Android settings:

![AndroidOtherSettings](https://user-images.githubusercontent.com/8617651/62394068-5d6c8800-b56c-11e9-9082-e2a11b89cdad.png)

**Comment by stowy** [comment]
If you change scripting backend to Mono does that change things?

**Comment by iseeall** [comment]
Nope, switching to Mono doesn't change it.

Actually I understand that one can't use Mono as scripting backend now because then ARM64 build can't be checked, and the ARM7 build causes a spike of crashes on about 4% of devices (error libmonobdwgc-2.mono_log_write_logcat)

![Crashes](https://user-images.githubusercontent.com/8617651/62394799-8ee65300-b56e-11e9-8e9b-117650ea074f.png)

**Comment by stowy** [comment]
Are you using the AdMob test ad units or your own ad units? Perhaps create
an issue on our forum at
https://groups.google.com/forum/#!categories/google-admob-ads-sdk/game-engines
- they will be able to help debug your issue with you efficiently.

**Comment by leoHN** [comment]
I'm getting same issue. Ads are failing to load. iOS too.

**Comment by IJEMIN** [comment]
Same here. I got a lot of bad reviews about app crash for android 8.1~9.0.
Every time I release a new build, crash percentage immediately sparks more than 50 times compared to previous versions. And then the graph becomes slowly stable.

And I got android 9.0 device which is not crashed, but it does not show the ad.

**Comment by iseeall** [comment]
> Same here. I got a lot of bad reviews about app crash for android 8.1~9.0.
> Every time I release a new build, crash percentage immediately sparks more than 50 times compared to previous versions. And then the graph becomes slowly stable.
> 
> And I got android 9.0 device which is not crashed, but it does not show the ad.
> 
> Unity 2019.1.12~2019.1.14 on Mac, IL2CPP with strip engine code + stripping level -low.

I ended up reverting to Unity 2018.2.21f and AdMob plugin 3.15.1. Nothing else seems to be stable.

**Comment by stowy** [comment]
@iseeall [this commit](https://github.com/googleads/googleads-mobile-unity/commit/a3255c2e707de089ff72d775fd68c78b688e2620#diff-9af848d5771305211424083c25bdef20) shows we updated from 15.0.1 to 17.2.0 version of Google AdMob Android SDK for 3.16. Can you try changing the version of the SDK in your GoogleMobileAdsDependencies.xml file from 15.0.1 to 17.2.0 and see if this reproduces the behavior?  

Additionally, did you test 2019.1.11f1, that is the version of Unity I've been using and haven't seen any issues.

**Comment by IJEMIN** [comment]
I guess this "exploding crash" problem for 64bit binary is related with these recent issues with admob

- https://groups.google.com/forum/#!searchin/google-admob-ads-sdk/crash$209.0|sort:date/google-admob-ads-sdk/cMXAu0pv4CQ/EbOdQApSCAAJ
- https://groups.google.com/forum/#!searchin/google-admob-ads-sdk/crash|sort:date/google-admob-ads-sdk/Q-edauZ27g8/wCstsRhFAAAJ
- https://groups.google.com/forum/#!category-topic/google-admob-ads-sdk/game-engines/4RuwbBlLh6I
- https://groups.google.com/forum/#!searchin/google-admob-ads-sdk/crash$209.0%7Csort:date/google-admob-ads-sdk/MK8de6JwqOg/rZbwPo5cEAAJ

**Comment by stowy** [comment]
@IJEMIN those seem to be various crashes related to the Android SDK and not specific to unity, or this issue.

**Comment by stowy** [comment]
I also ran a build from 2019.1.11f1 on my Pixel 3 - see [this thread](https://github.com/googleads/googleads-mobile-unity/issues/995#issuecomment-521714696) - Ads loading and displaying correctly with version 3.18.3.

**Comment by IJEMIN** [comment]
@stowy OK. I got it. I just want to let him know his, and my exploding crash problem might be related with recent web view issue with AdMob not unity.

@iseeall Maybe you can check normal info log message "Ad failed to load : 3" with "Ads" tagged in logcat. 

For me, while there are so many ads crashes on user devices, I got many "Ads fail load : 3" for my test devices which indicate ad request success but AdMob has no proper ad to show. And that's why I can't see the crash on my devices.

**Comment by M4YX0R** [comment]
Hello guys, same here. No ads, no logs, only crashes, problems and apps without ads. I have looked with logcat but there is nothing. If you want to build app bundle you should build with IL2CPP to built 64-bit variation but looks like Unity and Admob is not ready to use IL2CPP instead of mono.

**Comment by stowy** [comment]
Again, if reporting issues here, please specify which version of the Unity
IDE, Mobile Ads Plugin, Android API and device you are using.

I have tested with 2019.1.11f1 and the latest plugin version (3.18.3) on a
Pixel 3 using Android API 28 and I'm seeing ads load as expected. We are
seeing issues with version 2019.2 of Unity onwards at this point.

**Comment by stowy** [comment]
Regarding the above comment for issues in 2019.2, the issue was related to SMART_BANNER sized banner ads, and this has been fixed in version 4.0.0.

**Comment by stowy** [comment]
Closing obsolete issue.
